src/UI/Implementation/Render/ilJavaScriptBinding.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/* Copyright (c) 2016 Richard Klees <richard.klees@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Implementation\Render;
+
+/**
+ * Wraps global ilTemplate to provide JavaScriptBinding. 
+ */
+class ilJavaScriptBinding implements JavaScriptBinding {
+	/**
+	 * @var \ilTemplate
+	 */
+	private	$global_tpl;
+
+	/**
+	 * @var	integer
+	 */
+	private $id_count = 0;
```

Comment:
Don't mix client and server side. In one request on the server your point my be valid. But a given view might survive through async requests multiple such requests to the server and therefore multiple of your instances of JavaScriptBinding, leaving room for collisions.